source/extensions/path/uri_template_lib/uri_template.cc

```diff
@@ -0,0 +1,165 @@
+#include "source/extensions/path/uri_template_lib/uri_template.h"
+
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "source/common/http/path_utility.h"
+#include "source/extensions/path/uri_template_lib/proto/uri_template_rewrite_segments.pb.h"
+#include "source/extensions/path/uri_template_lib/uri_template_internal.h"
+
+#include "absl/status/statusor.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "re2/re2.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UriTemplate {
+
+using Internal::ParsedUrlPattern;
+
+#ifndef SWIG
+// Silence warnings about missing initializers for members of LazyRE2.
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif
+
+absl::StatusOr<std::string> convertURLPatternSyntaxToRegex(absl::string_view url_pattern) {
+  absl::StatusOr<ParsedUrlPattern> status = Internal::parseURLPatternSyntax(url_pattern);
```

Comment:
I think I want to keep as-is. It provides some separation from the internal implementation

Reply:
Ping

Reply:
Looking through.. I believe the reason this was done for to allow easier testing and usage by letting you pass in a string_view instead of the ParsedUrlPattern.